components/match2/wikis/clashroyale/match_group_input_custom.lua

```diff
@@ -0,0 +1,724 @@
+
+	for opponentIndex = 1, MAX_NUM_OPPONENTS do
+		local opponent = match['opponent' .. opponentIndex]
+		if Logic.isNotEmpty(opponent) then
+			--determine opponent scores, status and placement
+			--determine MATCH winner, resulttype and walkover
+			if String.isNotEmpty(walkover) then
+				if Logic.isNumeric(walkover) then
+					walkover = tonumber(walkover)
+					if walkover == opponentIndex then
+						match.winner = opponentIndex
+						match.walkover = 'L'
+						opponent.status = 'W'
+					elseif walkover == 0 then
+						match.winner = 0
+						match.walkover = 'L'
+						opponent.status = 'L'
+					else
+						local score = string.upper(opponent.score or '')
+						opponent.status = CONVERT_STATUS_INPUT[score] or 'L'
+					end
+				elseif Table.includes(ALLOWED_STATUSES, string.upper(walkover)) then
+					if tonumber(match.winner or 0) == opponentIndex then
+						opponent.status = 'W'
+					else
+						opponent.status = CONVERT_STATUS_INPUT[string.upper(walkover)] or 'L'
+					end
+				else
+					local score = string.upper(opponent.score or '')
+					opponent.status = CONVERT_STATUS_INPUT[score] or 'L'
+					match.walkover = 'L'
+				end
+				opponent.score = -1
+				match.finished = true
+				match.resulttype = 'default'
+			elseif CONVERT_STATUS_INPUT[string.upper(opponent.score or '')] then
+				if string.upper(opponent.score) == 'W' then
+					match.winner = opponentIndex
+					match.resulttype = 'default'
+					match.finished = true
+					opponent.score = -1
+					opponent.status = 'W'
+				else
+					match.resulttype = 'default'
+					match.finished = true
+					match.walkover = CONVERT_STATUS_INPUT[string.upper(opponent.score)]
+					local score = string.upper(opponent.score)
+					opponent.status = CONVERT_STATUS_INPUT[score]
+					opponent.score = -1
+				end
+			else
+				opponent.status = 'S'
+				opponent.score = tonumber(opponent.score) or
+					tonumber(opponent.sumscore) or -1
+				if opponent.score > bestof / 2 then
+					match.finished = Logic.emptyOr(match.finished, true)
+					match.winner = tonumber(match.winner or '') or opponentIndex
+				end
+			end
+
+			if Logic.readBool(match.cancelled) then
+				match.finished = true
+				if String.isEmpty(match.resulttype) and String.isEmpty(opponent.score) then
+					match.resulttype = 'np'
+					opponent.score = opponent.score or -1
+				end
+			end
+		else
+			break
+		end
+	end
+
+	CustomMatchGroupInput._determineWinnerIfMissing(match)
+
+	for opponentIndex = 1, MAX_NUM_OPPONENTS do
+		local opponent = match['opponent' .. opponentIndex]
+		if match.winner == 'draw' or tonumber(match.winner) == 0 or
+				(match.opponent1.score == bestof / 2 and match.opponent1.score == match.opponent2.score) then
+			match.finished = true
+			match.winner = 0
+			match.resulttype = 'draw'
+		end
+		if tonumber(match.winner) == opponentIndex or match.resulttype == 'draw' then
+			opponent.placement = 1
+		elseif Logic.isNumeric(match.winner) then
+			opponent.placement = 2
+		end
+	end
```

Comment:
Why is this its own iteration and not part of the one above?

Reply:
because there is a call to `CustomMatchGroupInput._determineWinnerIfMissing` between it
that function determines scores from maps and hence they are only available after it (unless manually entered or walkover or such applies)